app/public/js/localSprites.js

```diff
@@ -173,21 +201,39 @@ function updateLocalSprite(localSprite) {
-                if (localSprite.yForce) localSprite.target.vy += localSprite.yForce;
-                localSprite.cooldownTimer++;
+    //update triggers
+    if (localSprite.type === TRIGGER_TYPE_FORCE) {
+        if (rectanglesOverlap(localSprite.hitBox, getGlobalSpriteHitBox(localSprite.target)) && localSprites.length <2/*canTriggerTrigger(localSprite)*/) {
+            if (localSprite.forceType === FORCE_AMP) {
+                if (localSprite.target.vx) localSprite.target.vx *= localSprite.xForce;
+                if (localSprite.target.vy) localSprite.target.vy *= localSprite.yForce;              
+            }
+            if (localSprite.forceType === FORCE_FIXED) {
+                if (localSprite.target.vx) localSprite.target.vx += localSprite.xForce;
+                if (localSprite.target.vy) localSprite.target.vy += localSprite.yForce;
+            }
+            localSprite.notReadyToTriggerUntil = now() + localSprite.cooldownInMS;
+        }
+    }
+    
+    if (localSprite.type === TRIGGER_TYPE_SPAWN) {
+        if (rectanglesOverlap(localSprite.hitBox, getGlobalSpriteHitBox(localSprite.target)) && localSprites.length <2/*canTriggerTrigger(localSprite)*/) {
+            if (localSprite.spawnedObjectType === SPRITE_TYPE_HOMING_FIREBALL) addHomingFireballSprite(localSprite.spawnedObjectX, localSprite.spawnedObjectY, localSprite.target);
+            localSprite.notReadyToTriggerUntil = now() + localSprite.cooldownInMS;
+        }
+    }
+    if (localSprite.type === POWERUP_TYPE_HEART) {
+        if (rectanglesOverlap(localSprite.hitBox, getGlobalSpriteHitBox(mainCharacter))) {     //when I changed "localSprite.hitBox" to "getGlobalSpriteHitBox(localSprite), this stopped working.
+            if (localSprite.type === POWERUP_TYPE_HEART && mainCharacter.health < mainCharacter.maxHealth) {
+                mainCharacter.health++;
+                localSprite.shouldBeRemoved = true;
+            }
         }
-        if (localSprite.cooldownTimer > 0 && localSprite.cooldownTimer < localSprite.cooldownFrames) localSprite.cooldownTimer++;
-        else localSprite.cooldownTimer = 0;
     }
-    if (localSprite.type === 'SPRITE_TYPE_POWERUP') {
-        if (rectanglesOverlap(localSprite.hitBox, getGlobalSpriteHitBox(localSprite.target))) {     //when I changed "localSprite.hitBox" to "getGlobalSpriteHitBox(localSprite), this stopped working.
-            if (localSprite.powerupType === 'POWERUP_TYPE_HEART') {
-                localSprite.target.health++;
+    if (localSprite.type === POWERUP_TYPE_AIRDASH) {
+        if (rectanglesOverlap(localSprite.hitBox, getGlobalSpriteHitBox(mainCharacter))) {     //when I changed "localSprite.hitBox" to "getGlobalSpriteHitBox(localSprite), this stopped working.
+            if (localSprite.type === POWERUP_TYPE_AIRDASH) {
+                mainCharacter.canAirDashUntil = now() + localSprite.durationInMS;   //last number is ms
```

Comment:
I thin since it is called `durationInMS` you probably don't need the comment here any more. Also contextually since `now()` is milliseconds, it would be a mistake to add anything else to it.